### PR TITLE
Support for loading graph store data in lgraph (c++ end)

### LINKED ISSUE #952: Support for loading graph store data in c++ end
Currently, lgraph (c++ wrapper for rust graph store using ffis) can only query graph using graph handle set by jna and misses the following features:
1. Opening the graph store from store path directly.
2. Fetching the graph schema from graph store.
This severely limits the function of directly loading graph data on the C++ side.

### PR DESCRIPTION
**Pull request by GoldenLeaves**
- Open a graph store from path
- Get graph schema from opened store.
- Code reformat in lgraph
- Persist graph schema info when creating and restoring backups.

Fixes #952

### CODE REVIEW

**Comment by sighingnow** [comment]
OK to merge if we are blocked by the helm test failure.

File: interactive_engine/groot/src/main/java/com/alibaba/graphscope/groot/coordinator/NotifyFrontendListener.java
Comment (tianliplus) [inline interactive_engine/groot/src/main/java/com/alibaba/graphscope/groot/coordinator/NotifyFrontendListener.java:38]: It's not appropriate to put snapshot cache in every listener, since only one snapshot cache is needed in coordinator but there are many listeners here.

Comment (GoldenLeaves) [reply interactive_engine/groot/src/main/java/com/alibaba/graphscope/groot/coordinator/NotifyFrontendListener.java:38]: oka

Comment (GoldenLeaves) [reply interactive_engine/groot/src/main/java/com/alibaba/graphscope/groot/coordinator/NotifyFrontendListener.java]: oka

**Comment by yecol** [comment]
If possible, please address the issues raised by codefactor in the CI checks.

**Comment by GoldenLeaves** [comment]
> If possible, please address the issues raised by codefactor in the CI checks.

Oh okay, I thought it was because the ci task ran for a long time and failed because it timed out

**Comment by GoldenLeaves** [comment]
> If possible, please address the issues raised by codefactor in the CI checks.

CodeFactor issues fixed

**Review by LiSu — APPROVED** [review]
LGTM

**Pull request merged**
